wlog-zap/internal/provider.go

```diff
@@ -31,52 +30,32 @@ func LoggerProvider() wlog.LoggerProvider {
 type loggerProvider struct{}
 
 func (lp *loggerProvider) NewLogger(w io.Writer) wlog.Logger {
-	logger, atomicLevel := newZapLogger(w, wlog.InfoLevel, zapcore.EncoderConfig{
-		EncodeTime:     rfc3339NanoTimeEncoder,
-		EncodeDuration: zapcore.NanosDurationEncoder,
-	})
 	return &zapLogger{
-		logger: logger,
-		level:  atomicLevel,
+		logger: newZapLogger(w, zapcore.EncoderConfig{
+			EncodeTime:     rfc3339NanoTimeEncoder,
+			EncodeDuration: zapcore.NanosDurationEncoder,
+		}),
 	}
 }
 
 func (lp *loggerProvider) NewLeveledLogger(w io.Writer, level wlog.LogLevel) wlog.LeveledLogger {
-	logger, atomicLevel := newZapLogger(w, level, zapcore.EncoderConfig{
-		EncodeTime:     rfc3339NanoTimeEncoder,
-		EncodeDuration: zapcore.NanosDurationEncoder,
-		EncodeLevel:    zapcore.CapitalLevelEncoder,
-	})
 	return &zapLogger{
-		logger: logger,
-		level:  atomicLevel,
+		logger: newZapLogger(w, zapcore.EncoderConfig{
+			EncodeTime:     rfc3339NanoTimeEncoder,
+			EncodeDuration: zapcore.NanosDurationEncoder,
+			EncodeLevel:    zapcore.CapitalLevelEncoder,
+		}),
+		AtomicLogLevel: wlog.NewAtomicLogLevel(level),
 	}
 }
 
 func rfc3339NanoTimeEncoder(t time.Time, enc zapcore.PrimitiveArrayEncoder) {
 	enc.AppendString(t.In(time.UTC).Format(time.RFC3339Nano))
 }
 
-func newZapLogger(w io.Writer, logLevel wlog.LogLevel, encoderConfig zapcore.EncoderConfig) (*zap.Logger, *zap.AtomicLevel) {
-	level := zap.NewAtomicLevel()
-	level.SetLevel(toZapLevel(logLevel))
-	enc := zapcore.NewJSONEncoder(encoderConfig)
-	return zap.New(zapcore.NewCore(enc, zapcore.AddSync(w), level)), &level
-}
+func newZapLogger(w io.Writer, encoderConfig zapcore.EncoderConfig) *zap.Logger {
+	// We do our own enforcement in the level-specific methods; no need for zap to check again.
```

Comment:
nit: for consistency, please avoid "I/we" in comments -- update to something like "wlog performs its own enforcement of leveled logging in level-specific methods..."